src/cors/cors_middleware.ts

```diff
@@ -0,0 +1,191 @@
+import { Drash, ServerRequest, vary } from "../../deps.ts";
+
+type CorsMiddlewareConfig = {
+  origin?:
+    | ((
+    request: any,
```

Comment:
Not a problem here, but we plan on creating an interface for Drash's `request`, so if this PR gets merged ill make an issue for someone (you ideally :D) to update that type when we create Drash's request interface

Reply:
I can use ServerRequest from Deno for now, if its okay

Reply:
Does ServerRequest work when you use this middleware with Drash? I’m afraid there might be some weird side effects if it’s typed as ServerRequest because we use any. Would love to see tests exercise the implementation of this.

Reply:
im using the middleware in a deno app, without problems and using ServerRequest as type